### DISCUSSION TITLE
Settlement date lookup

### DISCUSSION THREAD

**Discussion by davidxbuck** [gerrymanoim/exchange_calendars#138] [Ideas]
I'm a fan of the library and the recent improvements, but I have one need that isn't currently met (unless you can tell me otherwise).

US markets have two days (Indigenous Peoples/Columbus Day and Veterans Day) where the markets are open the but banks are not and, as such, they are not settlement days. 

From [FINRA's regulations](https://www.finra.org/rules-guidance/notices/information-notice-101521):
>_The securities exchanges will be open for trading on Columbus Day and Veterans Day. Therefore, each day is considered a business day under Regulation T of the Federal Reserve Board, SEA Rule 15c3-3 and FINRA Rule 4210. However, please note each holiday is not a settlement date because banking institutions will be closed on those days._

As a result, I currently have manual processes that have to be put in place around those dates as I have no way to identify that a particular day is/isn't a settlement day.

I'm not sure if this is something that could possibly be in scope  (even a simple `is_settlement_date` method would be a bonus) but it would be useful to me and, quite likely, useful to others.

**Comment by gerrymanoim** [comment]
Thanks for the feedback! 

I think your best bet might be to create a new `EquitySettlementCalendar` that subclasses a base calendar (like https://github.com/gerrymanoim/exchange_calendars/blob/master/exchange_calendars/exchange_calendar_xnys.py#L76), but overrides the specific properties you want to modify with whatever holidays you want to include/exclude.

**Comment by maread99** [comment]
Alternatively (if @gerrymanoim were in agreement), you could offer a PR that adds what you're after as calendar-specific properties/methods of [XNYSExchangeCalendar](https://github.com/gerrymanoim/exchange_calendars/blob/master/exchange_calendars/exchange_calendar_xnys.py#L76).

The PR shouldn't change the behaviour of any existing properties/methods, rather only add what you require. Any new properties/methods should include tests and be well documented, including a note that they are specific to XNYS.

**Reply by gerrymanoim** [reply]
Yep! That definitely also works.

**Reply by davidxbuck** [reply]
Thanks for the prompt feedback, both. I'll look into this idea more deeply over the next few days.
